### ISSUE TITLE
Attempting free on address which was not malloc()-ed

### ISSUE THREAD

**Issue by mannol** [seppo0010/rlite#27]
Executing the following code:
```
// Open file database and setup a context c
rliteCommand(c, "SET a b");
rliteCommand(c, "DEL a"); // << removing this DEL and it works
rliteCommand(c, "SET a b");
```

Crashes with ASAN output:
```
=================================================================
==18629==ERROR: AddressSanitizer: attempting free on address which was not malloc()-ed: 0x60200000e090 in thread T0
    #0 0x4c9860 in __interceptor_cfree.localalias.0 (/home/mannol/Documents/Programming/Closenger/Debug-Desktop/Closenger+0x4c9860)
    #1 0x751894 in rl_btree_node_destroy /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/page_btree.c:125
    #2 0x729e8c in rl_write /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/rlite_internal.c:807
    #3 0x759c78 in rl_long_set /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/page_long.c:37
    #4 0x72a0e1 in rl_delete /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/rlite_internal.c:859
    #5 0x753026 in rl_btree_add_element /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/page_btree.c:432
    #6 0x70dd6f in rl_key_set /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/page_key.c:75
    #7 0x7205b7 in rl_set /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/type_string.c:50
    #8 0x7022ba in setGenericCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:2491
    #9 0x702620 in setCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:2534
    #10 0x6fa5d5 in rliteAppendCommandClient /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:1086
    #11 0x6fa78e in rlitevAppendCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:1119
    #12 0x6fa961 in rlitevCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:1150
    #13 0x6faa3d in rliteCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:1159
    #14 0x4fda3d in main /home/mannol/Documents/Programming/Closenger/Debug-Desktop/../main.cpp:69:5

0x60200000e090 is located 0 bytes inside of 54032-byte region [0x60200000e090,0x60200001b3a0)
==18629==AddressSanitizer CHECK failed: /build/llvm-toolchain-3.8-RMgQsv/llvm-toolchain-3.8-3.8.1/projects/compiler-rt/lib/asan/asan_allocator.cc:667 "((res.trace)) != (0)" (0x0, 0x0)
    #0 0x4d2f4d in __asan::AsanCheckFailed(char const*, int, char const*, unsigned long long, unsigned long long) (/home/mannol/Documents/Programming/Closenger/Debug-Desktop/Closenger+0x4d2f4d)
    #1 0x4d98e3 in __sanitizer::CheckFailed(char const*, int, char const*, unsigned long long, unsigned long long) (/home/mannol/Documents/Programming/Closenger/Debug-Desktop/Closenger+0x4d98e3)
    #2 0x42c3ac in __asan::AsanChunkView::GetAllocStack() (/home/mannol/Documents/Programming/Closenger/Debug-Desktop/Closenger+0x42c3ac)
    #3 0x4cdbb0 in __asan::DescribeHeapAddress(unsigned long, unsigned long) (/home/mannol/Documents/Programming/Closenger/Debug-Desktop/Closenger+0x4cdbb0)
    #4 0x4d0c31 in __asan::ReportFreeNotMalloced(unsigned long, __sanitizer::BufferedStackTrace*) (/home/mannol/Documents/Programming/Closenger/Debug-Desktop/Closenger+0x4d0c31)
    #5 0x42f1a5 in __asan::asan_free(void*, __sanitizer::BufferedStackTrace*, __asan::AllocType) (/home/mannol/Documents/Programming/Closenger/Debug-Desktop/Closenger+0x42f1a5)
    #6 0x4c983c in __interceptor_cfree.localalias.0 (/home/mannol/Documents/Programming/Closenger/Debug-Desktop/Closenger+0x4c983c)
    #7 0x751894 in rl_btree_node_destroy /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/page_btree.c:125
    #8 0x729e8c in rl_write /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/rlite_internal.c:807
    #9 0x759c78 in rl_long_set /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/page_long.c:37
    #10 0x72a0e1 in rl_delete /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/rlite_internal.c:859
    #11 0x753026 in rl_btree_add_element /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/page_btree.c:432
    #12 0x70dd6f in rl_key_set /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/page_key.c:75
    #13 0x7205b7 in rl_set /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/type_string.c:50
    #14 0x7022ba in setGenericCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:2491
    #15 0x702620 in setCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:2534
    #16 0x6fa5d5 in rliteAppendCommandClient /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:1086
    #17 0x6fa78e in rlitevAppendCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:1119
    #18 0x6fa961 in rlitevCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:1150
    #19 0x6faa3d in rliteCommand /home/mannol/Documents/Programming/Closenger/rlite/Debug-Desktop/../src/hirlite.c:1159
    #20 0x4fda3d in main /home/mannol/Documents/Programming/Closenger/Debug-Desktop/../main.cpp:69:5
```

**Comment by mannol** [comment]
It works on :memory: database...

**Comment by seppo0010** [comment]
Fixed in 4688a34ef9bb919b0f25ad30e590ce4285db907a. Thanks for the report.

**Comment by mannol** [comment]
Fix confirmed; thank you!
